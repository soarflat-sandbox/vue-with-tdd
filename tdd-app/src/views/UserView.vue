<template>
  <div>
    <VUserSearchForm />
    <VUserProfile />
  </div>
</template>

<script>
import VUserSearchForm from "@/components/VUserSearchForm";
import VUserProfile from "@/components/VUserProfile";
export default {
  name: "UserView",
  components: {
    VUserSearchForm,
    VUserProfile
  }
};
</script>
