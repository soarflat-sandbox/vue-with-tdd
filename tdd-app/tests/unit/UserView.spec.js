import { shallowMount } from "@vue/test-utils";
import UserView from "@/views/UserView";
import VUserSearchForm from "@/components/VUserSearchForm";
import VUserProfile from "@/components/VUserProfile";

describe("UserView", () => {
  it("コンポーネントがレンダリングされる", () => {
    const wrapper = shallowMount(UserView);

    // __snapshots__にhtmlのスナップショットが出力される
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("子コンポーネントがレンダリングされる", () => {
    const wrapper = shallowMount(UserView);
    // VUserSearchFormにマッチする`Wrapper`を返す
    const userSearchForm = wrapper.find(VUserSearchForm);
    // VUserProfileにマッチする`Wrapper`を返す
    const userProfile = wrapper.find(VUserProfile);

    // `Wrapper`（今回の場合、VUserSearchFormとVUserProfile）が存在すれば`true`を返す
    expect(userSearchForm.exists()).toBe(true);
    expect(userProfile.exists()).toBe(true);
  });
});
