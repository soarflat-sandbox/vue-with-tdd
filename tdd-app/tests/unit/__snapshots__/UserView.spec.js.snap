// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView コンポーネントがレンダリングされる 1`] = `
<div>
  <vusersearchform-stub></vusersearchform-stub>
  <vuserprofile-stub></vuserprofile-stub>
</div>
`;
